frontend/src/metabase/query_builder/containers/QueryBuilder.jsx

```diff
@@ -357,11 +357,9 @@ function QueryBuilder(props) {
     }
   }, [location, params, previousLocation, locationChanged]);
 
-  useEffect(() => {
-    if (question) {
-      question._update = updateQuestion;
-    }
-  });
+  if (question) {
+    question._update = updateQuestion;
+  }
```

Comment:
Moving this out of the `useEffect` ensures the function is attached earlier. It's gross, obviously, but it was already gross. I'd like to not do this at all soon // force any code dependent on `question.update(null, ...)` or `query.update()` to instead dispatch `updateQuestion` directly, since there aren't any other actions used with `_update`.